crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -859,7 +860,7 @@ pub struct CheckpointBuilder {
     notify: Arc<Notify>,
     notify_aggregator: Arc<Notify>,
     effects_store: Arc<dyn TransactionCacheRead>,
-    accumulator: Arc<StateAccumulator>,
+    accumulator: Weak<StateAccumulator>,
```

Comment:
Why is this Weak? We recreate checkpoint builder every epoch, right?

Reply:
See the comments between @mystenmark and I from a previous commit, but short answer is yes we tear it down at end of epoch, but this allows us to enforce the invariant in general that there are no dangling references to state accumulator when we tear it down and replace it. Mitigation against a bug creeping in if some other component later references state accumulator for example.

We can't make that guarantee if we have this remain an `Arc` because in that case we would race with checkpoint builder shutdown to count the strong references (checkpoint service is shutdown via an async message channel)